### PR TITLE
Process raw delay_period field as nanoseconds instead of seconds

### PR DESCRIPTION
**Pull request by soareschen**
Closes: #927

## Description

- Process raw `delay_period` field as nanoseconds instead of seconds.
- Use `Duration::default()` instead of `Duration::from_secs(0)` for empty duration.
- Use `Duration::new()` for construction from raw duration field with both seconds and nanoseconds.
- Print CLI output using `.as_secs_f64()` instead of `.as_secs()` to also show sub-second duration.

______

For contributor use:

- [x] Updated the __Unreleased__ section of [CHANGELOG.md](https://github.com/informalsystems/ibc-rs/blob/master/CHANGELOG.md) with the issue.
- [ ] If applicable: Unit tests written, added test to CI.
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [x] Updated relevant documentation (`docs/`) and code comments.
- [x] Re-reviewed `Files changed` in the Github PR explorer.

### CODE REVIEW

File: modules/src/ics03_connection/connection.rs [hunk 1]
```diff
@@ -31,7 +31,7 @@ impl Default for ConnectionEnd {
             client_id: Default::default(),
             counterparty: Default::default(),
             versions: vec![],
-            delay_period: Duration::from_secs(0),
+            delay_period: Duration::default(),
```
Comment (adizere) [inline modules/src/ics03_connection/connection.rs:34]: Nit: I'm not sure about this. Makes sense that `Duration::default()` would be a 0, but I have a slight preference for being explicit with the default value (unless it's a type we define ourselves, such as `ClientId`).

File: modules/src/ics07_tendermint/client_state.rs [hunk 2]
```diff
@@ -113,15 +113,15 @@ impl ClientState {
     /// Helper function to verify the upgrade client procedure.
     /// Resets all fields except the blockchain-specific ones.
     pub fn zero_custom_fields(mut client_state: Self) -> Self {
-        client_state.trusting_period = Duration::from_secs(0);
+        client_state.trusting_period = Duration::default();
```
Comment (adizere) [inline modules/src/ics07_tendermint/client_state.rs:116]: Same comment as I had for `ConnectionEnd`.

File: relayer-cli/src/commands/tx/connection.rs [hunk 3]
```diff
@@ -58,7 +58,7 @@ impl Runnable for TxRawConnInitCmd {
             self,
             |chains: ChainHandlePair| {
                 Connection {
-                    delay_period: Duration::from_secs(0),
```
Comment (adizere) [inline relayer-cli/src/commands/tx/connection.rs:61]: Maybe consider introducing a DEFAULT_PACKET_DELAY const in the codebase? Similar to how we have a [`MAX_PACKET_DELAY`](https://github.com/informalsystems/ibc-rs/blob/c05f9c16231216bcd3ab2e7ca07626eafc4c9d3c/relayer/src/connection.rs#L27).

File: modules/src/ics03_connection/connection.rs (hunk 1 above)
Comment (romac) [reply modules/src/ics03_connection/connection.rs:34]: Agreed, I too would prefer to be more explicit.

Comment (romac) [reply modules/src/ics03_connection/connection.rs:34]: Too bad [`Duration::ZERO`](https://doc.rust-lang.org/stable/std/time/struct.Duration.html#associatedconstant.ZERO) is nightly-only for now.

Comment (soareschen) [reply modules/src/ics03_connection/connection.rs]: Yeah `Duration::ZERO` is only in nightly. I have added a constant `ZERO_DURATION` in `ibc::timestamp` and use that instead.

My main issue with `Duration::from_secs(0)` is that it obscures the other use of `Duration::from_secs` during searching, and there can be potential inconsistency of using `Duration::from_secs(0)` and `Duration::from_nanos(0)` in different places.

File: relayer-cli/src/commands/tx/connection.rs (hunk 3 above)
Comment (soareschen) [reply relayer-cli/src/commands/tx/connection.rs:61]: I have instead defined `ZERO_DURATION`.

**Review by ancazamfir — APPROVED** [review]
LGTM, thanks @soareschen!

**Pull request merged**